packages/sdk-aws-kms-signer/index.ts

```diff
@@ -0,0 +1,49 @@
+import { CredentialSigner, KeyAssertion } from '@dfns/sdk'
+import { toBase64Url } from '@dfns/sdk/utils'
+import { KMSClient, SignCommand, SigningAlgorithmSpec } from "@aws-sdk/client-kms";
+
+
+export class AwsKmsKeySigner implements CredentialSigner<KeyAssertion> {
+
+  constructor(
+    private options: {
+      kmsKeyId: string
+      awsRegion: string
+      signingAlgorithm: SigningAlgorithmSpec
+      credId: string
+      appOrigin: string
+      crossOrigin?: boolean
+    }
+  ) { }
+
+  async sign(challenge: string): Promise<KeyAssertion> {
+    const clientData = Buffer.from(
+      JSON.stringify({
+        type: 'key.get',
+        challenge,
+        origin: this.options.appOrigin,
+        crossOrigin: this.options.crossOrigin ?? false,
+      })
+    )
+
+    const client = new KMSClient({ region: this.options.awsRegion })
```

Comment:
I think this can be moved to the constructor, that will allow for connection reuse for multiple signing requests.

Reply:
I updated this, let me know if it works given how the constructor operations flow or if I need to instantiate it directly in the constructor.